lib/presentation/profile/profile_route.dart

```diff
@@ -0,0 +1,682 @@
+import 'package:flutterapp/resources/app_dimens.dart';
+import 'package:flutterapp/resources/app_images.dart';
+import 'package:flutterapp/resources/app_messages.dart';
+import 'package:flutterapp/resources/app_text_letter_spacing.dart';
+import 'package:flutterapp/resources/app_theme.dart';
+import 'package:flutterapp/widgets/progress/arc_progress.dart';
+import 'package:flutterapp/resources/app_text_sizes.dart';
+import 'package:flutterapp/widgets/progress/linear_progress.dart';
+import 'package:flutterapp/presentation/profile/achievement_widget.dart';
+import 'package:flutterapp/widgets/text/uidisplay_widget.dart';
+import 'package:flutterapp/widgets/text/uitext_widget.dart';
+import 'package:flutter_staggered_grid_view/flutter_staggered_grid_view.dart';
+
+class ProfileRoute extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return _ProfileRouteState();
+  }
+}
+
+class _ProfileRouteState extends State<ProfileRoute> {
+  bool isExpanded = false;
+
+  @override
+  Widget build(BuildContext context) {
+    setStatusBarColor(AppColors.colorBlackGray,
+        statusBarIconBrightness: Brightness.light);
+    return _getProfileWidget();
+  }
+
+  Widget _getProfileWidget() {
+    if (isExpanded) {
+      return _getExpandedWidget();
+    } else {
+      return _getDefaultWidget();
+    }
+  }
+
+  Widget _getDefaultWidget() {
+    return Container(
+      color: AppColors.colorWhiteGray,
+      child: ListView(
+        children: [
+          Container(
+            decoration: BoxDecoration(
+              image: DecorationImage(
+                image: AssetImage(AppImages.profileHeader),
+                fit: BoxFit.fill,
+              ),
+            ),
+            child: Column(
+              children: [
+                SizedBox(height: ProfileDimens.screenTopMargin),
+                _UserProfileInfoWidget(),
+                SizedBox(height: ProfileDimens.achievementsMarginTop),
+                _AchievementsWidget(),
+                SizedBox(
+                    height: ProfileDimens.expandAchievementsButtonMarginTop),
+                _ExpandAchievementsLabelWidget(
+                  onExpandAchievementsPressed,
+                  false,
+                ),
+              ],
+            ),
+          ),
+          _JoinTheGameWidget(),
+          _CardWidget(
+            AppMessages.loyaltySystem.toUpperCase(),
+            AppImages.giftCards,
+            AppMessages.giftCards,
+            onGiftCardsPressed,
+            AppImages.bonus,
+            AppMessages.bonuses,
+            onBonusesPressed,
+          ),
+          _CardWidget(
+            AppMessages.setting.toUpperCase(),
+            AppImages.payment,
+            AppMessages.paymentMethods,
+            onPaymentMethodsPressed,
+            AppImages.signOut,
+            AppMessages.signOut,
+            onSignOutPressed,
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _getExpandedWidget() {
```

Comment:
Same as in the previous comment, can you please separate widgets a bit more, so there will be less inner wrappings in the one of main widgets on the screen?